### PR TITLE
Feat/sign in admin

### PR DESCRIPTION
**Pull request by Blaxwo**
## Describe your changes

- I created sign in page for admin.
- I fixed the way to receive error message.

## Issue ticket code (and/or) and link

- [Link to JIRA ticket](https://2024internship.atlassian.net/browse/SCRUM-50)

### **General**

- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [x] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [x] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file.
- [x] Used camelCase for variables and functions
- [x] Date and time formats are on the constants
- [x] Functions are public only if it's used outside the class
- [x] No hardcoded values
- [ ] covered by tests
- [x] Check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/).

### Frontend

- [x] Components and business logic are separated
- [x] Colors, Font Size, and Font Name is on the theme or in the constants
- [x] No text in the components, use i18n approach
- [x] No inline styles
- [x] Imports are absolute
- [x] Attach a screenshot if PR has visual changes.
зображення
зображення

### CODE REVIEW

File: src/pages/SignInAdminPage/SignInAdminForm/index.tsx [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+  password: string;
+}
+
+function isFetchBaseQueryError(error: unknown): error is FetchBaseQueryError {
+  return typeof error === 'object' && error != null && 'status' in error;
+}
+
+function isSerializedError(error: unknown): error is SerializedError {
+  return typeof error === 'object' && error != null && 'message' in error;
+}
```
Comment (lyudad) [inline src/pages/SignInAdminPage/SignInAdminForm/index.tsx]: I assume it should be used for any component? or a lot of them?

File: src/pages/SignInAdminPage/SignInAdminForm/index.tsx [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+interface IFormInput {
+  email: string;
+  password: string;
+}
+
+function isFetchBaseQueryError(error: unknown): error is FetchBaseQueryError {
+  return typeof error === 'object' && error != null && 'status' in error;
+}
+
+function isSerializedError(error: unknown): error is SerializedError {
```
Comment (lyudad) [inline src/pages/SignInAdminPage/SignInAdminForm/index.tsx]: why unknown?

File: src/pages/SignInAdminPage/SignInAdminForm/index.tsx (hunk 1 above)
Comment (Blaxwo) [reply src/pages/SignInAdminPage/SignInAdminForm/index.tsx]: Yes, I changed the way get error message so now this part doesn't repeat itself.

File: src/pages/SignInAdminPage/SignInAdminForm/index.tsx (hunk 2 above)
Comment (Blaxwo) [reply src/pages/SignInAdminPage/SignInAdminForm/index.tsx]: I fixed it, now there is no unknown.

**Review by lyudad — APPROVED** [review]

**Pull request merged**